xrpackage.js

```diff
@@ -1901,10 +1901,18 @@ export class XRPackage extends XRNode {
     return mainFile.response.body;
   }
   addFile(pathname, data = '', type = 'application/octet-stream') {
-    let bundle = new wbn.Bundle(this.data);
-    const builder = _cloneBundle(bundle, {
-      except: ['/' + pathname],
-    });
+    let bundle;
+    let builder;
+
+    if (this.data.byteLength) {
+      bundle = new wbn.Bundle(this.data);
+      builder = _cloneBundle(bundle, {
+        except: ['/' + pathname],
+      });
+    } else {
+      builder = new wbn.BundleBuilder(primaryUrl + '/manifest.json');
```

Comment:
I was wondering if we should throw an error here if the file being added isn't `manifest.json`, basically meaning that if you have an empty XRPackage, then the first file you add should be a `manifest.json`. 

Does that sound like something that would make sense?

Reply:
Hm, in general I think it would be good to have that test be performed _somewhere_... I'm not sure if forcing it to be the first file is best though.

Maybe a better option is an `isValid` method? Which verifies the presence of `manifest.json` and any other criteria that would make the package unloadable.

Reply:
I think we might need to do it at the beginning because the primary URL is the /manifest.json file and it looks like `new wbn.Bundle(this.data)` throws if the primary URL doesn't exist in the manifest?

Reply:
Unless we add a new method `addFiles` that accepts an array of files, which would be used when creating an XRPackage from scratch and we can do a check to make sure that one of those files is the manifest.json maybe?

Reply:
Or requiring the manifest data in the constructor if we're making an empty XRPackage possibly!

Reply:
I realised my first comment might have been confusing so just as an example, if I swap these two lines around:

https://github.com/webaverse/xrpackage/blob/8c55691a1a43ab3edb8727ed12ada2a9340aef6c/tests/dynamically-create-xrpk.js#L40-L41

this test fails because when it tries to add the manifest.json second, then the new if here:

https://github.com/webaverse/xrpackage/blob/8c55691a1a43ab3edb8727ed12ada2a9340aef6c/xrpackage.js#L1907-L1914

is true but `bundle = new wbn.Bundle(this.data);` throws an exception about a missing file for the primary URL

Reply:
Hm, I see what you're saying. You had all of the same thoughts I had above 👍 .

It doesn't feel right to me that the first file that needs to be added is `manifest.json` or the whole thing explodes, even if we throw an error for it.

Can we add an empty `manifest.json` if it didn't exist on the first add? Subsequent adds can of course override it, but it would make it so the order you add in doesn't crash in any case.

Also, the multi-file version of this kind of workflow is probably `compileRaw`, but both that API and `addFile` have their uses.

Reply:
That makes sense! I've pushed a change to add an empty `manifest.json` with `{}` if `addFile()` is called on an empty `XRPackage`, and the file being added isn't already a `manifest.json`.

Reply:
Awesome, thanks! This part looks good now.